keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -584,6 +586,142 @@ would allow us to delete the pod and trying again with a new one, but is not don
+spec:
+  # The node name indicates the node.
+  # Each driver on a node provides pools of devices for allocation,
+  # with unique device names inside each pool.
+  # Usually, but not necessarily, that pool name is the same as the
+  # node name.
+  nodeName: worker-1
+  poolName: worker-1
+  driverName: gpu.dra.example.com
+  sharedCapacity:
+  - name: gpu-0-shared-memory
+    capacity: 40Gi
+  - name: gpu-0-memory-block-0
+    capacity: 1
+  - name: gpu-0-memory-block-1
+    capacity: 1
+  - name: gpu-0-memory-block-2
+    capacity: 1
+  - name: gpu-0-memory-block-3
+    capacity: 1
+  devices:
+  - name: gpu-0
+    sharedCapacityConsumed:
+    - name: gpu-0-shared-memory
+      capacity: 40Gi
+    - name: gpu-0-memory-block-0
+      capacity: 1
+    - name: gpu-0-memory-block-1
+      capacity: 1
+    - name: gpu-0-memory-block-2
+      capacity: 1
+    - name: gpu-0-memory-block-3
+      capacity: 1
+    attributes:
+    ...
+  - name: gpu-0-first-half
+    sharedCapacityConsumed:
+    - name: gpu-0-shared-memory
+      capacity: 20Gi
+    - name: gpu-0-memory-block-0
+      capacity: 1
+    - name: gpu-0-memory-block-1
+      capacity: 1
+    attributes:
+    ...
+  - name: gpu-0-middle-half
+    sharedCapacityConsumed:
+    - name: gpu-0-shared-memory
+      capacity: 20Gi
+    - name: gpu-0-memory-block-1
+      capacity: 1
+    - name: gpu-0-memory-block-2
+      capacity: 1
+    attributes:
+    ...
+  - name: gpu-0-second-half
+    sharedCapacityConsumed:
+    - name: gpu-0-shared-memory
+      capacity: 20Gi
+    - name: gpu-0-memory-block-2
+      capacity: 1
+    - name: gpu-0-memory-block-3
+      capacity: 1
+    attributes:
+    ...
+  - name: gpu-0-first-quarter
+    sharedCapacityConsumed:
+    - name: gpu-0-shared-memory
+      capacity: 10Gi
+    - name: gpu-0-memory-block-0
+      capacity: 1
+    attributes:
+    ...
+  - name: gpu-0-second-quarter
+    sharedCapacityConsumed:
+    - name: gpu-0-shared-memory
+      capacity: 10Gi
+    - name: gpu-0-memory-block-1
+      capacity: 1
+    attributes:
+    ...
+  - name: gpu-0-third-quarter
+    sharedCapacityConsumed:
+    - name: gpu-0-shared-memory
+      capacity: 10Gi
+    - name: gpu-0-memory-block-2
+      capacity: 1
+    attributes:
+    ...
+  - name: gpu-0-fourth-quarter
+    sharedCapacityConsumed:
+    - name: gpu-0-shared-memory
+      capacity: 10Gi
+    - name: gpu-0-memory-block-3
+      capacity: 1
+    attributes:
+    ...
+```
+
+In this example, `gpu-0-first-half` and `gpu-0-second-half` could be allocated
+simultaneouly (because the set of `gpu-0-memory-block`s they consume are
+mutually exclusive). However, `gpu-0-first-half` and `gpu-0-first-quarter`
+could not (because `gpu-0-memory-block-0` is consumed by both of them).
+
+To enable this feature, scheduler support will need to be added to track the
+set of `SharedCapacity` objects provided by each `ResourceSlice` and ensure
+that no two devices are ever allocated that consume overlapping capacities.
```

Comment:
```suggestion
that the set of allocated devices does not exceed the available `SharedCapacity`.
```

"overlapping" sounds very specific to MIG. I think it's easier to reason about this model when focusing on the mechanism (sum up consumed resources, compare against total).

Reply:
Updated the text with more details